packages/core/helper-plugin/lib/src/hooks/useLibrary/index.js

```diff
@@ -0,0 +1,20 @@
+/**
+ *
+ * useLibrary
+ *
+ */
+
+import { useContext, useRef } from 'react';
+import LibraryContext from '../../contexts/LibraryContext';
+
+const useLibrary = () => {
+  const { components, fields } = useContext(LibraryContext);
+  // Use a ref so we can safely add the components or the fields
+  // to a hook dependencies array
+  const composRef = useRef(components);
+  const fieldsRef = useRef(fields);
```

Comment:
I would prefer to avoid this kind of patterns. Anyways, the component using the useLibrary hook will rerender everytime components or fields change.

Then it's up to us to create valid useEffect so we don't end up in situations where references become a problem 😊

Reply:
> Anyways, the component using the useLibrary hook will rerender everytime components or fields change.

You can't change the library after the app is rendered